skrf/noisyNetwork.py

```diff
@@ -0,0 +1,812 @@
+========================================
+
+Provides an n-port network class that holds noise covariance matrices. This
+is useful when you want to calculate the noise figure and noise parameters
+of a network. 
+
+:class:`NoisyNetwork` derives from :class:`Network`. Therefore, any methods
+within the :class:`Network` class can be used from :class:`NoisyNetwork`. 
+For example, *resample*, *interpolate*, and any of the plotting functions
+within :class:`Network` and be used on this object.
+
+NoisyNetwork Class
+==================
+
+.. autosummary::
+    :toctree: generated/
+
+    NoisyNetwork
+
+Set Noise Source
+================
+*noise_source* is the primary method that sets a :class:`NoisyNetwork`
+apart from a :class:`Network`. This function is used to associate a noise
+covariance matrix with a network.
+
+.. autosummary::
+    :toctree: generated/
+
+    NoisyNetwork.noise_source
+
+Noise Parameters
+================
+.. autosummary::
+    :toctree: generated/
+
+    NoisyNetwork.y_opt
+    NoisyNetwork.z_opt
+    NoisyNetwork.g_opt
+    NoisyNetwork.rn
+    NoisyNetwork.nfmin
+    NoisyNetwork.nfmin_db
+    NoisyNetwork.nf
+    NoisyNetwork.nf_db
+
+Network Noise Covariance Representations
+========================================
+.. autosummary::
+    :toctree: generated/
+
+    NoisyNetwork.cs
+    NoisyNetwork.ct
+    NoisyNetwork.cz
+    NoisyNetwork.cy
+    NoisyNetwork.ca
+
+Network Methods
+---------------
+These functions are part of the parent class :class:`Network` and are 
+provided here for convenience.
+
+.. autosummary::
+    :toctree: generated/
+
+    Network.from_z
+    Network.from_y
+    Network.from_a
+    Network.s
+    Network.z
+    Network.y
+    Network.a
+    Network.t
+
+Connecting Networks with Noise Analysis
+=======================================
+
+.. autosummary::
+    :toctree: generated/
+
+    cascade_2port
+    parallel_parallel_2port
+    series_series_2port
+
+
+"""
+
+from six.moves import xrange
+from six import string_types
+
+
+import numpy as npy
+from copy import deepcopy as copy
+
+from . import mathFunctions as mf
+from .network import Network
+from .network import z2a, z2s, z2t, z2y, y2a, y2s, y2t, y2z, s2a, s2h, s2t, s2y
+from .network import  s2z, a2s, a2t, a2z, t2a, t2s, t2y, t2z
+from .frequency import Frequency
+from .networkNoiseCov import NetworkNoiseCov
+
+from .constants import ZERO, K_BOLTZMANN, T0
+from .constants import S_DEFINITIONS, S_DEF_DEFAULT
+
+
+class NoisyNetwork(Network):
+    """
+    A :class:`Network` with associated noise covariance matrix and methods.
+
+    For instructions on how to create a NoisyNetwork see  :func:`__init__`.
+
+    =====================  =============================================
+    Operator               Function
+    =====================  =============================================
+    \+                     combines noisy networks in series
+    \|                     combines noisy networks in parallel
```

Comment:
these operators have a different meaning in `Network`. I'm afraid this be a source of future confusion and issues

Reply:
I agree. Would you be willing to permit adding a global flag that would enable changing the functionality? I did this before and it worked well. Basically you do something like rf.enable_new_operators() at the top of a script and it changes the functionality of the operators. What I like about using operators with two-port networks is that you can create an entire circuit and the result just looks like a line of algebra. This is far more convenient that having to call individual functions for each combination. 

But, again the ball is in your court. If you think this is too awkward I can be convince to drop this functionality. Or perhaps use different operators (if there are any left :D).

Reply:
It's an interesting feature, Is it working for Nport as well ?